### DISCUSSION TITLE
Mirth Connect 4.1.1 service stuck in stopping state after network disconnected and loss DB connection

### DISCUSSION THREAD

**Discussion by ZuheirF-PH** [nextgenhealthcare/connect#6160] [Q&A]
Hi, after upgrade Mirth Connect from 3.5.2 to 4.1.1 I am facing and issue when stopping the service after losing connection to DB,
the service stack in Stopping state. 
[Shutdown Hook Thread] com.mirth.connect.server.Mirth: shutting down mirth due to normal request
[Shutdown Hook Thread] com.mirth.connect.server.Mirth: stopping engine
[Shutdown Hook Thread] com.mirth.connect.server.controllers.DefaultConfigurationController: retrieving property: category=core, name=channelDependencies
[Shutdown Hook Thread] org.apache.ibatis.transaction.jdbc.JdbcTransaction: Openning JDBC Connection

I tried the scenario with 3.5.2 version and there not reproducible

thanks
Zuheir

**Comment by jonbartels** [comment]
> I tried the scenario with 3.5.2 version and there not reproducible

What is the full scenario? 

What is logged before this log entry? `[Shutdown Hook Thread] com.mirth.connect.server.Mirth: shutting down mirth due to normal request`

**Reply by ZuheirF-PH** [reply]
I have a cluster that the mirth set in one node and the db in other node, when I am executing failover that I am disconnecting the public network the cluster manager stop Mirth service and the service not stopped and stuck in Stopping state.

seems its issue with the HikariCP I changed the pool to DBCP and works now.
Is known issue in HikariCP?

**Reply by pacmano1** [reply]
Do you have the advanced cluttering plug-in or is this your own clustering solution?

**Reply by ZuheirF-PH** [reply]
Using Veritas as cluster manager,  but even we disable the network and stop the Mirth manually it's stuck.
